lib/ui/platform_dispatcher.dart

```diff
@@ -2605,3 +2621,37 @@ class SemanticsActionEvent {
     );
   }
 }
+
+/// Callback to subscribe to [ViewFocusChange] notifications.
+typedef ViewFocusChangeCallback = void Function(ViewFocusChange);
+
+/// An event to communicate view focus changes between the framework and engine.
+///
+/// If the event comes from the framework it means there was a flutter focus change.
+/// If the event comes from the engne it means there was a platform focus change.
+final class ViewFocusChange {
+  /// Creates a [ViewFocusChange].
+  const ViewFocusChange({
+    required this.viewId,
+    required this.viewFocusState,
+  });
+
+  /// The id of the [FlutterView] that experienced the focus change.
+  final int viewId;
+
+  /// The current [FocusState] of the view.
+  final ViewFocusState viewFocusState;
```

Comment:
I think it's sufficient to name this field simply `state` as both "view" and "focus" are implied from the context.

Reply:
Agree, makes sense.

Reply:
PTAL I changed names a bit so I now kinda feel "viewFocusState" makes sense.